### PR TITLE
Workflow - Build Conda Package

### LINKED ISSUE #1824: GitHub Actions - Workflow - build_conda_pkg
GitHub Actions workflow for build_conda_pkg.

Quip Doc: https://alteryx.quip.com/YbZzAp4EURNj/Workflow-buildcondapkg

### PR DESCRIPTION
**Pull request by ParthivNaresh**
Fixes #1824

### CODE REVIEW

File: evalml/tests/dependency_update_check/latest_dependency_versions.txt [hunk 1]
```diff
@@ -23,7 +23,7 @@ scikit-optimize==0.8.1
 scipy==1.6.2
 seaborn==0.11.1
 shap==0.39.0
-sktime==0.5.3
+sktime==0.6.0
```
Comment (jeremyliweishih) [inline evalml/tests/dependency_update_check/latest_dependency_versions.txt:26]: is this dependency change for this PR?

Comment (jeremyliweishih) [reply evalml/tests/dependency_update_check/latest_dependency_versions.txt:26]: oo this was for the dependency test right? I just merged in the PR so this isn't needed anymore 😄

Comment (ParthivNaresh) [reply evalml/tests/dependency_update_check/latest_dependency_versions.txt:26]: No this was unrelated but needed to pass the dependency check, sktime had a release a few hours ago and it doesn't break any of our code but I thought I should include it for bug fixes etc.

Comment (ParthivNaresh) [reply evalml/tests/dependency_update_check/latest_dependency_versions.txt:26]: ^^Ignore this haha I didn't see your next comment

File: .github/workflows/build_conda_pkg.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+name: Build Conda Package
+
+on:
+  pull_request:
```
Comment (freddyaboulton) [inline .github/workflows/build_conda_pkg.yml:4]: I think we only want this to run on pushes to main

**Review by freddyaboulton — APPROVED** [review]
@ParthivNaresh I think this looks good!

File: .github/workflows/build_conda_pkg.yml (hunk 2 above)
Comment (ParthivNaresh) [reply .github/workflows/build_conda_pkg.yml:4]: @freddyaboulton Currently `build_conda_pkg` runs in CircleCI as part of the other workflows, so on every commit, primarily for the `shellcheck`. Are we changing this behaviour @dsherry ?

Comment (freddyaboulton) [reply .github/workflows/build_conda_pkg.yml:4]: Ah, I think I missed `if: github.ref == 'refs/head/main'` line below. If that means that the package is only built on commits to main, then I think we're good to go!

But I think we can make it simpler by using `on: push: branches: -main`? I don't think we need the shellcheck to run on PR branches but let me know if I'm misunderstanding!

Comment (ParthivNaresh) [reply .github/workflows/build_conda_pkg.yml:4]: Got it, I was under the impression we were keeping the same behaviour as the CircleCI implementation but if we're good to not do a shellcheck on PR branches then I can definitely make the change to only run on `main`. I agree that I don't think it's necessary.

Comment (dsherry) [reply .github/workflows/build_conda_pkg.yml]: Sounds good to me!

I also wonder if shellcheck would belong better in our `lint` jobs.

File: .github/workflows/build_conda_pkg.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+name: Build Conda Package
+
+on:
+  push:
+    branches:
+      - main
```
Comment (dsherry) [inline .github/workflows/build_conda_pkg.yml:6]: @ParthivNaresh could you please add
```
  pull_request:
    types: [opened, synchronize]
```
and show that this action runs green on this PR before merge?

We've used that syntax in other actions, and I think it means the job will run in this PR but not in other PRs against `main`.

**Review by dsherry — CHANGES_REQUESTED** [review]
Looks great! One minor change requested, to prove that the action is green

**Review by angela97lin — APPROVED** [review]
Goooodbye codecov!! LGTM 🥂

File: .github/workflows/build_conda_pkg.yml (hunk 3 above)
Comment (dsherry) [reply .github/workflows/build_conda_pkg.yml:6]: Hey, my bad, I clearly do not know enough of the GH actions syntax yet 😆  I was hoping there was a way to run the action in this PR just to prove that it works before merging. If this is too difficult to do (which I think is the case after 5min of googling), no worries, let's merge it and revert if there are problems.

**Review by dsherry — APPROVED** [review]
🚢 😁

**Pull request merged**